### ISSUE TITLE
Add GA for tracking social sharing

### ISSUE THREAD

**Issue by mikehenrty** [mozilla/voice-web#772]

**Comment by flyingpimonster** [comment]
I'm not sure I like the idea of adding Google Analytics. The whole point of Common Voice is to *not* be dependent on big data companies like Google. Can you explain how this would benefit the project? Are there no viable alternatives?

**Comment by jf99** [comment]
I understand both sites. It looks a bit contradictory to use Amazon Web Services and Google Analytics on the one hand and on the other hand develop a tool to get away from them. One could argue, we beat them with their own weapons. But I tend to endorse @flyingpimonster 's point. Mozilla should have the resources to get along without those data leeches.

**Comment by paskalito** [comment]
why would we need to track social sharing???

if we would do social tracking.. - wich already seems super against mozilla philosophy - why would we need GA?

**Comment by mikehenrty** [comment]
Hi all, thanks for discussing this important topic.

To clarify, we already use Google Analytics in Common Voice. For a list of the information we track through it, see here:
https://github.com/mozilla/voice-web/blob/master/docs/data_dictionary.md

Google Analytics is used by most web properties at Mozilla. It has been thoroughly examined by our legal and privacy team, and we have found it meets our standards. That said, we do indeed respect the Do Not Track browser setting, and will not use GA if you have it set. The following are some links that might be helpful to learn about our policies:

https://www.mozilla.org/en-US/privacy/principles/
https://www.mozilla.org/en-US/privacy/websites/

Also, I have looked into using a self-hosted system like [Piwik](https://discourse.mozilla.org/t/piwik-instead-of-google-analytics/17326), but unfortunately our project does not have the resources to maintain this and at the same time build features Common Voice (we only have 1 full-time developer). In the future, I'd like to re-examine this.

Now, I could easily see this conversation turning into a philosophical thread about what tools Mozilla should and should not be using. First, let me say that this is not the right place for that, as it will have no effect on official Mozilla policy towards GA. Second, I urge you to remember that while it may seem like Mozilla has unlimited resources to always build and deploy things the "right" way (whatever that may mean), in reality Mozilla must be pragmatic and choose the right tools for the right job. That doesn't mean ignoring our values. Quite the contrary, we make every decision with our values in mind. Instead it means weighing all options and making the appropriate choice for the situation. Many times, that means using services like GA, AWS, etc.

And also let me clarify that Common Voice is in no way a tool for moving away from GA and AWS. It is about giving speech developers and users more choice when it comes to speech technology.

So to sum up, this bug is not about adding Google Analytics. We already have that. It's only about adding events for when someone uses the sharing links at the bottom of the page so we can know if that feature is useful to our users.
